src/content/app/regulatory-activity-viewer/components/region-overview/region-overview-image/region-overview-location-selector/useLocationSelector.ts

```diff
@@ -0,0 +1,199 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useReducer, useRef, type MutableRefObject } from 'react';
+
+/**
+ * TODO:
+ * - consider how this selection will be reconciled with selection start and end
+ *   that are passed from the parent (i.e. those would inevitably have to live in redux)
+ * - until selection has been committed, allow dismissing it upon pressing the escape key
+ */
+
+/**
+ * Actions:
+ * - start selection
+ * - update selection
+ * - clear selection
+ */
+
+type State = {
+  originX: number;
+  latestX: number;
+} | null;
+
+type SelectionStartAction = {
+  type: 'selection-start';
+  payload: {
+    originX: number;
+    latestX: number;
+  };
+};
+
+type SelectionUpdateAction = {
+  type: 'selection-update';
+  payload: {
+    x: number;
+  };
+};
+
+type SelectionClearAction = {
+  type: 'selection-clear';
+};
+
+type Action =
+  | SelectionStartAction
+  | SelectionUpdateAction
+  | SelectionClearAction;
+
+const selectionStateReducer = (state: State, action: Action): State => {
+  if (action.type === 'selection-start') {
+    return {
+      originX: action.payload.originX,
+      latestX: action.payload.latestX
+    };
+  } else if (action.type === 'selection-update') {
+    if (!state) {
+      // this should not happen
+      return null;
+    }
+
+    return {
+      ...state,
+      latestX: action.payload.x
+    };
+  } else {
+    return null;
+  }
+};
+
+const useLocationSelector = <T extends HTMLElement | SVGSVGElement>({
+  ref,
+  onSelectionCompleted
+}: {
+  ref: MutableRefObject<T | null>;
+  onSelectionCompleted: (coords: { start: number; end: number }) => void;
+}) => {
+  const [state, dispatch] = useReducer(selectionStateReducer, null);
+  const stateRef = useRef(state);
+  const selectionOriginXRef = useRef<number | null>(null);
+  const containerBoundingClientRef = useRef<DOMRect | null>(null);
+
+  useEffect(() => {
+    if (ref.current) {
+      containerBoundingClientRef.current = ref.current.getBoundingClientRect();
+
+      ref.current.addEventListener('mousedown', onMouseDown as EventListener);
+      // ref.current.addEventListener('touchstart', onSelectionStart as EventListener);
+    }
+
+    return () => {
+      if (!ref.current) {
+        return;
+      }
+      ref.current.removeEventListener(
+        'mousedown',
+        onMouseDown as EventListener
+      );
+      // ref.current.removeEventListener('touchstart', onSelectionStart as EventListener);
```

Comment:
Should this and the comment above be removed?

Reply:
I left these commented out touchstarts as a todo for testing this on a touch device. I can add a "TODO" comment; but also we can just leave this as it is.